ipynb/metadata_template_generator.md

```diff
@@ -0,0 +1,129 @@
+
+Environmental packages available:
+
+* air
+* built environment
+* host-associated
+* human-associated
+* human-skin
+* human-oral
+* human-gut
+* human-vaginal
+* microbial mat/biofilm
+* misc environment
+* plant-associated
+* sediment
+* soil
+* wastewater/sludge
+* water
+
+### Import libraries
+
+
+```python
+import pandas as pd
+import re
+```
+
+### Define variables
+
+
+```python
+# required fields for Qiita, EBI, and MIMS (general)
+qiita_ebi_mims_path = '../data/MIxS/Qiita_EBI_MIMS_v1.xlsx'
+
+# required fields for environmental packages
+migs_mims_path = '../data/MIxS/MIGS_MIMS_v4.xls'
+
+# list one environmental package e.g. soil or multiple e.g. soil,water (one recommended)
+list_of_env_pkg = 'soil'
+
+# number of samples for specified environmental package
+number_of_samples = 10
+
+# PI name followed by study number
+sample_prefix = 'Jones99'
+
+# used to pre-populate template
+investigation_type = 'metagenome'
+```
+
+### Function: return dataframe of environmental package-specific metadata items
+
+
+```python
+# a single environmental package (soil) or list can be provided (soil,water)
+def show_items_of_env_pkg(df_env_pkg, list_of_env_pkg):
+    df_items = df_env_pkg[df_env_pkg['Environmental package'].isin(list_of_env_pkg)]
+    return df_items
+```
+
+### Function: return dataframe of metadata template
+
+
+```python
+def create_template_for_env_pkg(df_QiitaEBI, df_MIMS, df_env_pkg, list_of_env_pkg, number_of_samples, sample_prefix):
+    
+    # get headers/requirement/example of Qiita-EBI/MIMS/env_pkg columns
+    pkg_items = show_items_of_env_pkg(df_env_pkg, list_of_env_pkg)
+    headers_env_pkg = pkg_items['Structured comment name'].values
+    require_env_pkg = pkg_items['Requirement']
+    example_env_pkg = pkg_items['Value syntax']
+    headers_all = list(df_QiitaEBI.iloc[0]) + list(df_MIMS.iloc[0]) + list(headers_env_pkg)
+    require_all = pd.concat([df_QiitaEBI.iloc[1], df_MIMS.iloc[1], require_env_pkg])
+    example_all = pd.concat([df_QiitaEBI.iloc[2], df_MIMS.iloc[2], example_env_pkg])
+    
+    # populate template dataframe
+    df_template = pd.DataFrame(columns=headers_all, dtype=object)
+    df_template.loc['Requirement'] = require_all.values
+    df_template.loc['Format'] = example_all.values
+    string_of_env_pkg = re.sub(r'\W', '.', '.'.join(list_of_env_pkg))
+    for i in range(0, number_of_samples):
+        df_template.loc[i+1] = ['' for x in range(len(df_template.columns))]
+        df_template.loc[i+1]['sample_name'] = '%s.%s.%s' % (sample_prefix, string_of_env_pkg, i+1)
+        df_template.loc[i+1]['investigation_type'] = investigation_type
+        df_template.loc[i+1]['env_package'] = ' or '.join(list_of_env_pkg)
+    return df_template
+```
+
+### Main function: generate metadata template and readme csv files
+
+
+```python
+def generate_metadata_template(qiita_ebi_mims_path, migs_mims_path, list_of_env_pkg, number_of_samples, sample_prefix):
+    
```

Comment:
[Numpy documenation standards](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

Its to clarify what the function does, what parameters are being pass into the function and what's being returned.